qiskit_optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -147,10 +148,26 @@ def __init__(
                 :class:`~qiskit_optimization.converters.QuadraticProgramToQubo` will be used.
 
         Raises:
+            TypeError: If minimum eigensolver has an invalid type.
             TypeError: When one of converters has an invalid type.
             QiskitOptimizationError: When the minimum eigensolver does not return an eigenstate.
         """
-
+        if isinstance(min_eigen_solver, VQE):
+            raise TypeError(
+                "MinimumEigenOptimizer does not support VQE but SamplingVQE. "
+                "qiskit.algorithms.minimum_eigensolvers.SamplingVQE may help you."
```

Comment:
```suggestion
                "MinimumEigenOptimizer does not support this VQE. You can use  "
                "qiskit.algorithms.minimum_eigensolvers.SamplingVQE instead."
```

Since it had SamplingVQE in short and in full following one from one another I altered to just leave the full one and changed text a little to suggest to make it more explicit they can use this instead.